frontend/src/components/widgets/Form/FormClearHelper.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { isValidFormId } from "src/lib/utils"
+import { WidgetStateManager } from "src/lib/WidgetStateManager"
+import { SignalConnection } from "typed-signals"
+
+export class FormClearHelper {
+  private formClearListener?: SignalConnection
+
+  private lastWidgetMgr?: WidgetStateManager
+
+  private lastFormId?: string
+
+  /**
+   * Register a listener that will be called when the widget's form is cleared.
+   * This should be called in the `render` function of every class-based widget
+   * element - it mimics the behavior of a `useEffect` hook, and ensures that
+   * subscription and unsubscription happen correctly.
+   *
+   * Hooks-based widgets can just use `useEffect` and call
+   * `widgetMgr.addFormClearedListener` directly.
+   */
+  public useFormClearListener(
```

Comment:
I don't think we should use `useFoo` naming here. It mimics a hook, but it isn't one.

Reply:
How about `addFormClearListener`?

Reply:
I changed it to `manageFormClearListener`